hudi-hadoop-common/src/main/java/org/apache/hudi/hadoop/fs/HoodieRetryWrapperFileSystem.java

```diff
@@ -185,12 +186,20 @@ public boolean rename(Path src, Path dst) throws IOException {
   @Override
   public boolean delete(Path f, boolean recursive) throws IOException {
     return new RetryHelper<Boolean, IOException>(maxRetryIntervalMs, maxRetryNumbers, initialRetryIntervalMs, retryExceptionsList)
-        .tryWith(() -> fileSystem.delete(f, recursive)).start().booleanValue();
+        .tryWith(() -> {
+          boolean success = fileSystem.delete(f, recursive);
+          if (!success) {
+            if (fileSystem.exists(f)) {
+              throw new HoodieIOException("Failed to delete file: " + f);
+            }
+          }
+          return success;
+        }).start().booleanValue();
   }
 
   @Override
   public boolean delete(Path f) throws IOException {
-    return new RetryHelper<Boolean, IOException>(maxRetryIntervalMs, maxRetryNumbers, initialRetryIntervalMs, retryExceptionsList).tryWith(() -> fileSystem.delete(f, true)).start().booleanValue();
+    return delete(f, true);
```

Comment:
Any other places in the repo that may not use `delete` APIs in these two classes?